src/helpers/format.ts

```diff
@@ -486,11 +498,15 @@ export function createDefaultFormat(value: number): Format {
   return decimalDigits ? "0." + "0".repeat(decimalDigits.length) : "0";
 }
 
-export function detectFormat(content: string, locale: Locale): Format | undefined {
-  if (isDateTime(content, locale)) {
-    const internalDate = parseDateTime(content, locale)!;
-    return internalDate.format;
+export function detectDateFormat(content: string, locale: Locale): Format | undefined {
+  if (!isDateTime(content, locale)) {
+    return undefined;
   }
+  const internalDate = parseDateTime(content, locale)!;
+  return internalDate.format;
+}
+
+export function detectNumberFormat(content: string, locale: Locale): Format | undefined {
```

Comment:
this is always used with `locale` being `DEFAULT_LOCALE` and it looks like it cannot work otherwise given the hard-coded `content.includes(".")`
Either remove the `locale` arguments and hard-code `DEFAULT_LOCALE` in the function or use the locale's separator instead of the "."